keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1328 @@
+typically be wide-open for reading, but locked-down for writing.
+
+Each ClusterTrustBundle object is a container for multiple named trust anchor
+sets.  Most use cases will simply use a single trust anchor set named `default`,
+but mutual TLS use cases involving multiple federating meshes will require the
+ability to express multiple, non-commingled trust anchor sets.
+
+ClusterTrustBundles may optionally be associated with a signer using their
+`.spec.signerName` field.  ClusterTrustBundles that are associated with a signer
+have special handling of creates and updates, restricting mutating actions to
+those with an RBAC grant on the signer name (similar to the handling of
+CertificateSigningRequest objects).
+
+To enable consumption by workloads, the new `trustAnchors` Kubelet projected
+volume source allows projecting a trust anchor set from a particular signer into
+the container filesystem, with the contents of the projected files updating as
+the corresponding trust anchor sets are updated.
+
+In general, ClusterTrustBundle objects are considered publicly-readable within
+the cluster.
+
+### User Stories
+
+#### Trust Anchor Distribution for Private CAs
+
+In my cluster, I operate a certificates/v1 signer implementation called
+`example.com/server-tls`.  This is backed by a private CA hierarchy.  By some
+unspecified mechanism, I issue certificates from this CA hierarchy to a server
+workload (a pod named `server/server`).  My client workload (a pod named
+`client/client`) needs access to the trust anchors for the CA in order to
+connect to the server.  The client should be able to continually establish
+connections to the server without downtime, even as I rotate the trust anchors
+that back the CA.
+
+To do this, I update the controller that implements `example.com/server-tls` to
+create a ClusterTrustBundle object like this one:
+```yaml
+apiVersion: v1alpha1
+kind: ClusterTrustBundle
+metadata:
+  name: example.com--server-tls-live
+spec:
+  signerName: example.com/server-tls
+status:
+  sets:
+  - name: default
+    pemTrustAnchors:
+    - "<... PEM DATA ...>"
+  conditions:
+  - type: "Ready"
+    status: "True"
+```
+
+As the set of trust anchors for my CA changes due to rotations and revocations,
+my controller updates the status of the ClusterTrustBundle object.
+
+I then update my client pod to add a `trustAnchors` volume:
+```diff
+apiVersion: v1
+kind: Pod
+metadata:
+  namespace: client
+  name: client
+spec:
+  containers:
+    name: main
+    image: my-image
+    volumeMounts:
++    - mountPath: /var/run/example-com-server-tls-trust-anchors
++      name: example-com-server-tls-trust-anchors
++      readOnly: true
+  volumes:
++  - name: example-com-server-tls-trust-anchors
++    projected:
++      sources:
++      - trustAnchors:
++          files:
++          - flatFile:
++              signerName: example.com/server-tls
++              setName: default
```

Comment:
How does the client learn these names?

Reply:
I don't think we have a better answer right now than "by reading the documentation for the signer you want to use".